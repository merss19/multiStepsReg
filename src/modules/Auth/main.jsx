"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) if (e.indexOf(p[i]) < 0)
            t[p[i]] = s[p[i]];
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_redux_1 = require("react-redux");
var react_router_1 = require("react-router");
var react_router_dom_1 = require("react-router-dom");
var ducks_1 = require("./ducks");
var CustomModal_1 = require("modules/CustomModal");
var SignUp_1 = require("modules/SignUp");
var storage_1 = require("tools/storage");
var selectors_1 = require("./selectors");
var mapStateToProps = function (state, ownProps) { return ({
    isFetching: selectors_1.selectIsFetching(state),
    isLoad: selectors_1.selectIsLoad(state),
    isAuth: selectors_1.selectIsAuth(state),
    isError: selectors_1.selectIsError(state),
    errMsg: selectors_1.selectErrMsg(state),
    user: selectors_1.selectUser(state)
}); };
var ProtectedRouteComponent = /** @class */ (function (_super) {
    __extends(ProtectedRouteComponent, _super);
    function ProtectedRouteComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ProtectedRouteComponent.prototype.componentWillMount = function () {
        var _this = this;
        var checkAuth = this.props.checkAuth;
        checkAuth(function (data) { return _this.checkAuth(data); }, function () { return _this.leave(); });
    };
    ProtectedRouteComponent.prototype.componentWillReceiveProps = function (nextProps) {
        if (nextProps.isFetching !== this.props.isFetching ||
            nextProps.isError !== this.props.isError) {
            var isOpen = nextProps.isFetching || nextProps.isError;
            this.props.toggleModal(isOpen);
        }
    };
    ProtectedRouteComponent.prototype.leave = function () {
        var _a = this.props, history = _a.history, toggleModal = _a.toggleModal;
        toggleModal(false);
        storage_1.storage.remove('token', { path: '/' });
        storage_1.storage.remove('txId', { path: '/' });
        storage_1.storage.remove('program', { path: '/' });
        storage_1.storage.remove('packageType', { path: '/' });
        storage_1.storage.remove('promoName', { path: '/' });
        history.replace('/');
    };
    ProtectedRouteComponent.prototype.checkAuth = function (data) {
        var _a = this.props, changeStep = _a.changeStep, history = _a.history;
        if (data.paidState !== 0) {
            storage_1.storage.save('userProgram', data.program, { path: '/', maxAge: 60 * 60 * 24 * 365 * 10 });
            if (data.paidState === 2) {
                changeStep(SignUp_1.Steps.four);
                history.replace('/signup');
            }
            else if (data.isFirstEdit) {
                history.replace('/profile');
            }
            else if (!data.emailConfirmed) {
                return null;
            }
            else {
                history.replace('/task');
            }
        }
        else if (!data.program) {
            changeStep(SignUp_1.Steps.two);
        }
        else {
            changeStep(SignUp_1.Steps.four);
        }
    };
    ProtectedRouteComponent.prototype.modalClick = function () {
        var _a = this.props, history = _a.history, toggleModal = _a.toggleModal;
        toggleModal(false);
        history.replace('/');
    };
    ProtectedRouteComponent.prototype.clickHandler = function () {
        var isError = this.props.isError;
        if (!isError) {
            return null;
        }
        return this.modalClick();
    };
    ProtectedRouteComponent.prototype.renderModal = function (modal) {
        var _a = this.props, isFetching = _a.isFetching, errMsg = _a.errMsg, isError = _a.isError;
        return (<CustomModal_1.CustomModal modal={modal} resultText={errMsg} clickHandler={this.clickHandler.bind(this)} isLoader={isFetching}/>);
    };
    ProtectedRouteComponent.prototype.renderContent = function (props) {
        var _a = this.props, ProtectedComponent = _a.component, isFetching = _a.isFetching, errMsg = _a.errMsg;
        return (<div>
        <ProtectedComponent {...props}/>
        {this.renderModal(CustomModal_1.ModalTypes.error)}
      </div>);
    };
    ProtectedRouteComponent.prototype.renderRoute = function (props) {
        var isAuth = this.props.isAuth;
        return (<div>
        {isAuth ?
            this.renderContent(props) :
            this.renderModal(CustomModal_1.ModalTypes.info)}
      </div>);
    };
    ProtectedRouteComponent.prototype.render = function () {
        var _this = this;
        var _a = this.props, ProtectedComponent = _a.component, isAuth = _a.isAuth, isError = _a.isError, isFetching = _a.isFetching, errMsg = _a.errMsg, rest = __rest(_a, ["component", "isAuth", "isError", "isFetching", "errMsg"]);
        return (<react_router_dom_1.Route {...rest} render={function (props) {
            return _this.renderRoute(props);
        }}/>);
    };
    return ProtectedRouteComponent;
}(React.Component));
exports.ProtectedRouteComponent = ProtectedRouteComponent;
exports.ProtectedRoute = react_redux_1.connect(mapStateToProps, {
    checkAuth: ducks_1.checkAuth,
    changeStep: SignUp_1.changeStep,
    toggleModal: CustomModal_1.toggleModal
})(ProtectedRouteComponent);
exports.default = react_router_1.withRouter(exports.ProtectedRoute);
