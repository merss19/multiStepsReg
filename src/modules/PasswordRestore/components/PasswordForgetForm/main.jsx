"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var redux_form_1 = require("redux-form");
var react_router_dom_1 = require("react-router-dom");
var Button_1 = require("components/Button");
var InputText_1 = require("components/InputText");
var formValidations = require("tools/validations");
var PasswordForgetForm = /** @class */ (function (_super) {
    __extends(PasswordForgetForm, _super);
    function PasswordForgetForm() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    PasswordForgetForm.prototype.render = function () {
        var _a = this.props, handleSubmit = _a.handleSubmit, onSubmit = _a.onSubmit;
        return (<div className='entry__inner'>
        <div className='entry__header'>
          <h2 className='entry__title text-center'>Восстановление пароля</h2>
        </div>
        <form onSubmit={handleSubmit(onSubmit)} className='entry__box'>
          <redux_form_1.Field name='email' id='email' placeholder='Ваш email' component={InputText_1.default}/>
          <Button_1.Button styleBtn={Button_1.ButtonTypes.success} type='submit' wide={true}>
            Восстановить пароль
          </Button_1.Button>
          <ul className='entry__link text-center'>
            <li className='entry-nav__item'>
              <react_router_dom_1.Link to='/'>Вход в кабинет</react_router_dom_1.Link>
            </li>
          </ul>
        </form>
      </div>);
    };
    return PasswordForgetForm;
}(React.PureComponent));
exports.PasswordForgetForm = PasswordForgetForm;
exports.default = redux_form_1.reduxForm({
    form: 'PasswordForm',
    validate: formValidations.validateForgetPassword
})(PasswordForgetForm);
