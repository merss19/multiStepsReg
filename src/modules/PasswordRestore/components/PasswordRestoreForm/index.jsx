"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var main_1 = require("./main");
exports.default = main_1.default;
