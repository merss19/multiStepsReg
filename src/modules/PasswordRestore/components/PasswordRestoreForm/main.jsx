"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var redux_form_1 = require("redux-form");
var react_router_dom_1 = require("react-router-dom");
var Button_1 = require("components/Button");
var InputText_1 = require("components/InputText");
var formValidations = require("tools/validations");
var PasswordRestoreForm = /** @class */ (function (_super) {
    __extends(PasswordRestoreForm, _super);
    function PasswordRestoreForm() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    PasswordRestoreForm.prototype.render = function () {
        var _a = this.props, handleSubmit = _a.handleSubmit, onSubmit = _a.onSubmit;
        return (<div className='entry__inner'>
        <div className='entry__header'>
          <h2 className='entry__title text-center'>Восстановление пароля</h2>
        </div>
        <form onSubmit={handleSubmit(onSubmit)} className='entry__box'>
          <redux_form_1.Field name='pass' id='pass' placeholder='Новый пароль' component={InputText_1.default}/>
          <redux_form_1.Field name='passAgain' id='passAgain' placeholder='Новый пароль повторно' component={InputText_1.default}/>

          <Button_1.Button styleBtn={Button_1.ButtonTypes.info} type='submit' wide={true}>
            Восстановить пароль
          </Button_1.Button>
          <ul className='entry__link text-center'>
            <li className='entry-nav__item'>
              <react_router_dom_1.Link to='/signup'>Регистрация</react_router_dom_1.Link>
            </li>
          </ul>
        </form>
      </div>);
    };
    return PasswordRestoreForm;
}(React.PureComponent));
exports.PasswordRestoreForm = PasswordRestoreForm;
exports.default = redux_form_1.reduxForm({
    form: 'RestoreForm',
    validate: formValidations.validatePasswordRestore
})(PasswordRestoreForm);
