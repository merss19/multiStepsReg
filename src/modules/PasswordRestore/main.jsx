"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_redux_1 = require("react-redux");
var react_router_1 = require("react-router");
var PasswordForgetForm_1 = require("./components/PasswordForgetForm");
var PasswordRestoreForm_1 = require("./components/PasswordRestoreForm");
var ducks_1 = require("./ducks");
var CustomModal_1 = require("../CustomModal");
var LayoutAddHOC_1 = require("components/HOC/LayoutAddHOC");
var utils_1 = require("tools/utils");
var selectors_1 = require("./selectors");
var mapStateToProps = function (state, OwnProps) { return ({
    isFetching: selectors_1.selectIsFetching(state),
    token: selectors_1.selectToken(state),
    isLoad: selectors_1.selectIsLoad(state),
    isError: selectors_1.selectIsError(state),
    resultText: selectors_1.selectResultText(state),
    resultCode: selectors_1.selectResultCode(state),
    errorMessage: selectors_1.selectErrorMessage(state)
}); };
var PasswordRestoreComponent = /** @class */ (function (_super) {
    __extends(PasswordRestoreComponent, _super);
    function PasswordRestoreComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    PasswordRestoreComponent.prototype.componentWillMount = function () {
        var _a = this.props, location = _a.location, setRestoreToken = _a.setRestoreToken;
        var query = {};
        if (location.search) {
            console.log('111111111111111111');
            query = utils_1.queryParse(location.search);
            if (query.token) {
                console.log('222222222222222222222');
                setRestoreToken(query.token);
            }
        }
    };
    PasswordRestoreComponent.prototype.componentWillReceiveProps = function (nextProps) {
        if (nextProps.isFetching !== this.props.isFetching ||
            nextProps.isError !== this.props.isError ||
            nextProps.isLoad !== this.props.isLoad) {
            var isOpen = nextProps.isFetching || nextProps.isError || nextProps.isLoad;
            this.props.toggleModal(isOpen);
        }
    };
    PasswordRestoreComponent.prototype.onSubmit = function (data) {
        var submitPassword = this.props.submitPassword;
        submitPassword(data);
    };
    PasswordRestoreComponent.prototype.onSubmitRestore = function (data) {
        var _a = this.props, submitRestore = _a.submitRestore, token = _a.token;
        data.tokenPassword = token;
        submitRestore(data);
    };
    PasswordRestoreComponent.prototype.setModal = function () {
        var _a = this.props, isLoad = _a.isLoad, isError = _a.isError, resultCode = _a.resultCode;
        var modal = CustomModal_1.ModalTypes.info;
        if (isError) {
            modal = CustomModal_1.ModalTypes.error;
        }
        if (isLoad) {
            if (resultCode === 1) {
                modal = CustomModal_1.ModalTypes.success;
            }
        }
        return modal;
    };
    PasswordRestoreComponent.prototype.clickHandler = function () {
        var _a = this.props, token = _a.token, history = _a.history;
        var modal = this.setModal();
        // console.log('clickkkkkkkkkkk')
        // console.log(token && modal === ModalTypes.success)
        if (token && modal === CustomModal_1.ModalTypes.success) {
            console.log('!!!!!!');
            history.replace('/');
        }
    };
    PasswordRestoreComponent.prototype.render = function () {
        var _a = this.props, isFetching = _a.isFetching, token = _a.token, isError = _a.isError, resultText = _a.resultText, errorMessage = _a.errorMessage;
        var modal = this.setModal();
        var text = isError ? errorMessage : resultText;
        return (<div className='entry--min'>
        {!token
            ?
                <PasswordForgetForm_1.default onSubmit={this.onSubmit.bind(this)}/>
            :
                <PasswordRestoreForm_1.default onSubmit={this.onSubmitRestore.bind(this)}/>}
        <CustomModal_1.CustomModal modal={modal} resultText={text} clickHandler={this.clickHandler.bind(this)} isLoader={isFetching}/>
      </div>);
    };
    return PasswordRestoreComponent;
}(React.PureComponent));
exports.PasswordRestoreComponent = PasswordRestoreComponent;
exports.PasswordRestore = react_redux_1.connect(mapStateToProps, {
    submitPassword: ducks_1.submitPassword,
    submitRestore: ducks_1.submitRestore,
    setRestoreToken: ducks_1.setRestoreToken,
    toggleModal: CustomModal_1.toggleModal
})(PasswordRestoreComponent);
exports.default = LayoutAddHOC_1.LayoutAddHOC()(react_router_1.withRouter(exports.PasswordRestore));
