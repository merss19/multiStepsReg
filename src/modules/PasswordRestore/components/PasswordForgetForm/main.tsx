import * as React from 'react';
import {Field, reduxForm, FormProps} from 'redux-form';
import {Link} from 'react-router-dom';
import {PasswordFormData} from '../../interfaces';
import {Button, ButtonTypes} from 'components/Button';
import InputText from 'components/InputText';
import * as formValidations from 'tools/validations';

interface OwnProps extends FormProps<PasswordFormData, {}, {}> {
  onSubmit: (values: PasswordFormData) => void
}

export class PasswordForgetForm extends React.PureComponent<OwnProps, {}> {

  render() {
    const {handleSubmit, onSubmit} = this.props;
    return (
      <div className='entry__inner'>
        <div className='entry__header'>
          <h2 className='entry__title text-center'>Восстановление пароля</h2>
        </div>
        <form onSubmit={handleSubmit(onSubmit)} className='entry__box'>
          <Field
            name='email'
            id='email'
            placeholder='Ваш email'
            component={InputText}
          />
          <Button
            styleBtn={ButtonTypes.success}
            type='submit'
            wide={true}
          >
            Восстановить пароль
          </Button>
          <ul className='entry__link text-center'>
            <li className='entry-nav__item'>
              <Link to='/'>Вход в кабинет</Link>
            </li>
          </ul>
        </form>
      </div>
    )
  }
}

export default reduxForm({
  form: 'PasswordForm',
  validate: formValidations.validateForgetPassword
})(PasswordForgetForm);