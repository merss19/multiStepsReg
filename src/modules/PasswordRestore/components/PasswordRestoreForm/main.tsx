import * as React from 'react';
import {Field, reduxForm, FormProps} from 'redux-form';
import {Link} from 'react-router-dom';
import {RestoreFormData} from '../../interfaces';
import {Button, ButtonTypes} from 'components/Button';
import InputText from 'components/InputText';
import * as formValidations from 'tools/validations';

interface OwnProps extends FormProps<RestoreFormData, {}, {}> {
  onSubmit: (values: RestoreFormData) => void
}

export class PasswordRestoreForm extends React.PureComponent<OwnProps, {}> {

  render() {
    const {handleSubmit, onSubmit} = this.props;
    return (
      <div className='entry__inner'>
        <div className='entry__header'>
          <h2 className='entry__title text-center'>Восстановление пароля</h2>
        </div>
        <form onSubmit={handleSubmit!(onSubmit)} className='entry__box'>
          <Field name='pass' id='pass' placeholder='Новый пароль' component={InputText}/>
          <Field name='passAgain' id='passAgain' placeholder='Новый пароль повторно' component={InputText}/>

          <Button
            styleBtn={ButtonTypes.info}
            type='submit'
            wide={true}
          >
            Восстановить пароль
          </Button>
          <ul className='entry__link text-center'>
            <li className='entry-nav__item'>
              <Link to='/signup'>Регистрация</Link>
            </li>
          </ul>
        </form>
      </div>
    );
  }
}

export default reduxForm({
  form: 'RestoreForm',
  validate: formValidations.validatePasswordRestore
})(PasswordRestoreForm);