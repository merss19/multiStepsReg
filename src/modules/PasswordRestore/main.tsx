import * as React from 'react';
import {connect} from 'react-redux';
import {withRouter, RouteComponentProps} from 'react-router';
import PasswordForgetForm from './components/PasswordForgetForm';
import PasswordRestoreForm from './components/PasswordRestoreForm';
import {PasswordFormData, RestoreFormData} from './interfaces';
import {submitPassword, submitRestore, setRestoreToken} from './ducks';
import {CustomModal, toggleModal, ModalType, ModalTypes} from '../CustomModal';
import {LayoutHOC} from 'components/HOC/LayoutHOC';
import {queryParse} from "tools/utils";

import {
  selectIsFetching,
  selectIsLoad,
  selectToken,
  selectIsError,
  selectResultText,
  selectErrorMessage,
  selectResultCode
} from './selectors';


type OwnProps = {}

type ConnectedDispatch = {
  submitPassword(data: PasswordFormData): void
  submitRestore(data: RestoreFormData): void
  toggleModal(isOpen: boolean): void
  setRestoreToken(data: string): void
}
type ConnectedState = {
  isFetching: boolean
  isLoad: boolean
  isError: boolean
  resultText: string
  token: string
  resultCode: number
  errorMessage: string
}
const mapStateToProps = (state: any, OwnProps: OwnProps): ConnectedState => ({
  isFetching: selectIsFetching(state),
  token: selectToken(state),
  isLoad: selectIsLoad(state),
  isError: selectIsError(state),
  resultText: selectResultText(state),
  resultCode: selectResultCode(state),
  errorMessage: selectErrorMessage(state)
});
type props = RouteComponentProps<string> & ConnectedState & ConnectedDispatch & OwnProps;

export class PasswordRestoreComponent extends React.PureComponent<props, {}> {

  componentWillMount() {
    const {location, setRestoreToken} = this.props;
    let query: any = {};

    if(location.search){
      console.log('111111111111111111')
      query = queryParse(location.search);
      if(query.token){
        console.log('222222222222222222222')
        setRestoreToken(query.token);
      }
    }
  }

  componentWillReceiveProps(nextProps: props) {
    if (
      nextProps.isFetching !== this.props.isFetching ||
      nextProps.isError !== this.props.isError ||
      nextProps.isLoad !== this.props.isLoad) {
      const isOpen: boolean = nextProps.isFetching || nextProps.isError || nextProps.isLoad;
      this.props.toggleModal(isOpen);
    }
  }

  onSubmit(data: PasswordFormData) {
    const {submitPassword} = this.props;
    submitPassword(data);
  }

  onSubmitRestore(data: RestoreFormData) {
    const {submitRestore, token} = this.props;
    data.tokenPassword = token;
    submitRestore(data);
  }

  setModal() {
    const {isLoad, isError, resultCode} = this.props;
    let modal: ModalType = ModalTypes.info;

    if (isError) {
      modal = ModalTypes.error;
    }
    if (isLoad) {
      if (resultCode === 1) {
        modal = ModalTypes.success;
      }
    }
    return modal;
  }

  clickHandler() {
    const {token, history} = this.props;
    const modal: ModalType = this.setModal();
   // console.log('clickkkkkkkkkkk')
   // console.log(token && modal === ModalTypes.success)
    if (token && modal === ModalTypes.success) {
      console.log('!!!!!!')
      history.replace('/')
    }
  }

  render() {
    const {
      isFetching,
      token,
      isError,
      resultText,
      errorMessage
    } = this.props;
    const modal: ModalType = this.setModal();
    const text: string = isError ? errorMessage : resultText;

    return (
      <div className='entry--min'>
        {!token
          ?
          <PasswordForgetForm onSubmit={this.onSubmit.bind(this)}/>
          :
          <PasswordRestoreForm onSubmit={this.onSubmitRestore.bind(this)}/>
        }
        <CustomModal
          modal={modal}
          resultText={text}
          clickHandler={this.clickHandler.bind(this)}
          isLoader={isFetching}
        />
      </div>
    )
  }
}

export const PasswordRestore = connect(mapStateToProps, {
  submitPassword,
  submitRestore,
  setRestoreToken,
  toggleModal
})(PasswordRestoreComponent);

export default LayoutHOC()(withRouter(PasswordRestore));