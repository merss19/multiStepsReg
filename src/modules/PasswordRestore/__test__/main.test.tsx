import {shallow, mount} from 'enzyme';
import {PasswordRestoreComponent, PasswordRestore} from '../main';
import PasswordForgetForm from '../components/PasswordForgetForm';
import PasswordRestoreForm from '../components/PasswordRestoreForm';
import * as React from 'react';
import toJson from 'enzyme-to-json';
import * as sinon from 'sinon';
import {Provider} from 'react-redux';
import configureStore from '../../../store';
import {BrowserRouter as Router} from 'react-router-dom';
import {ModalTypes} from '../../CustomModal';
import {configure} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import {queryParse} from "../../../tools/utils/index";

const store = configureStore();
configure({adapter: new Adapter()});

let data = {
  pass: 'qeqwe',
  passAgain: 'qweqwe',
  tokenPassword: 'asdasdasdasd'
};
let payload = {
  email: 'test@email.ru'
};
let changeStep,
  submitLogin,
  toggleModal,
  onSubmit,
  submitPassword,
  submitRestore,
  setRestoreToken,
  setToken,
  setError,
  isFetching,
  isLoad,
  isError,
  errMsg,
  token,
  resultText,
  resultCode,
  errorMessage,
  match,
  location,
  history;
const start = () => {
  beforeEach(() => {
    changeStep = jest.fn(),
      toggleModal = jest.fn(),
      setToken = jest.fn(),
      onSubmit = jest.fn(),
      submitPassword = jest.fn(),
      submitRestore = jest.fn(),
      setRestoreToken = jest.fn(),
      isFetching = false,
      isLoad = false,
      isError = false,
      errMsg = 'Ошибка',
      token = 'token',
      resultText = 'resultText',
      resultCode = 1,
      errorMessage = 'errorMessage',
      match = {},
      location = {
        search: ''
      };
    history = {
      replace: jest.fn()
    }
  });
  const props = {
    submitPassword: submitPassword,
    submitRestore: submitRestore,
    setRestoreToken: setRestoreToken,
    toggleModal: toggleModal,
    setToken: setToken,
    setError: setError,
    isFetching: isFetching,
    isLoad: isLoad,
    isError: isError,
    errMsg: errMsg,
    match: match,
    location: location,
    history: history,
    token: token,
    resultText: resultText,
    resultCode: resultCode,
    errorMessage: errorMessage
  };
  const buildComponent = () => {
    return shallow(<PasswordRestoreComponent {...props}/>)
  };
  const buildComponentMount = () => {
    return mount(
      <Provider store={store}>
        <Router>
          <PasswordRestore {...props}/>
        </Router>
      </Provider>)
  };
  return {
    buildComponentMount,
    buildComponent
  }
};
describe('Component PasswordRestore', () => {
  start();
  it('renders component', () => {
    const component = start().buildComponent();
    expect(component).toHaveLength(1);
  });
  it('renders correctly', () => {
    const tree = toJson(start().buildComponent());
    expect(tree).toMatchSnapshot();
  });
  it('renders component Mount', () => {
    const component = start().buildComponentMount();
    expect(component).toHaveLength(1);
  });
  it('call setRestoreToken,  location.search === null', () => {
    const component = start().buildComponent();
    let query: any = queryParse(component.instance().props.location.search);
    expect(query).toBeNull();
    expect(setRestoreToken).not.toHaveBeenCalled();
  });
  it('call setRestoreToken, location.search !== null', () => {
    location = {
      search: '?token=asdasd'
    };
    const component = start().buildComponent();
    let query: any = queryParse(component.instance().props.location.search);
    expect(setRestoreToken).toHaveBeenCalledWith(query.token);
  });
  it('call setRestoreToken, location.search === null', () => {
    location = {
      search: '?token'
    };
    const component = start().buildComponent();
    let query: any = queryParse(component.instance().props.location.search);
    expect(query.token).toBeUndefined();
    expect(setRestoreToken).not.toHaveBeenCalled();
  });
  it('call onSubmit - call submitPassword', () => {
    const component = start().buildComponent();
    const instance = component.instance() as PasswordRestoreComponent;
    instance.onSubmit(payload);
    expect(submitPassword).toHaveBeenCalledWith(payload);
  });
  it('call onSubmitRestore - call submitRestore', () => {
    const component = start().buildComponent();
    const instance = component.instance() as PasswordRestoreComponent;
    instance.onSubmitRestore(data);
    expect(submitRestore).toHaveBeenCalledWith(data);
  });
  it('render PasswordRestoreForm', () => {
    const component = start().buildComponent();
    expect(component.find(PasswordRestoreForm)).toHaveLength(1);
    expect(component.find(PasswordForgetForm)).toHaveLength(0);
  });
  it('render PasswordForgetForm', () => {
    token = null;
    const component = start().buildComponent();
    expect(component.find(PasswordForgetForm)).toHaveLength(1);
    expect(component.find(PasswordRestoreForm)).toHaveLength(0);
  });
  it('call clickHandler', () => {
    const component = start().buildComponent();
    const spy = sinon.spy(PasswordRestoreComponent.prototype, 'setModal');
    const instance = component.instance() as PasswordRestoreComponent;
    instance.clickHandler();
    expect(spy.called).toEqual(true)
  });
  it('call clickHandler - do not call location.replace(): modal!== success', () => {
    const component = start().buildComponent();
    const instance = component.instance() as PasswordRestoreComponent;
    instance.clickHandler();
    expect(history.replace).not.toHaveBeenCalled();
  });
  it('call clickHandler - do not call location.replace(): modal === error', () => {
    isError = true;
    const component = start().buildComponent();
    const instance = component.instance() as PasswordRestoreComponent;
    instance.clickHandler();
    expect(history.replace).not.toHaveBeenCalled();
  });
  it('call clickHandler - do not call location.replace(): token = null', () => {
    isLoad = true;
    token = null;
    const component = start().buildComponent();
    const instance = component.instance() as PasswordRestoreComponent;
    instance.clickHandler();
    expect(history.replace).not.toHaveBeenCalled();
  });
  it('call clickHandler - call location.replace(): modal === success', () => {
    isLoad = true;
    const component = start().buildComponent();
    const instance = component.instance() as PasswordRestoreComponent;
    instance.clickHandler();
    expect(history.replace).toHaveBeenCalledWith('/');
  });
  describe('componentWillReceiveProps', () => {
    start();
    it('do not call toggleModal', () => {
      const component = start().buildComponent();
      expect(toggleModal).not.toHaveBeenCalled();
      token = 'error';
      isFetching = false;
      component.setProps({token: token, isFetching: isFetching});
      expect(toggleModal).not.toHaveBeenCalled();
    });
    it('call toggleModal isFetching = true', () => {
      const component = start().buildComponent();
      isFetching = true;
      component.setProps({isFetching: isFetching});
      expect(toggleModal).toHaveBeenCalledWith(true);
    });
    it('call toggleModal isLoad = true', () => {
      const component = start().buildComponent();
      isLoad = true;
      component.setProps({isLoad: isLoad});
      expect(toggleModal).toHaveBeenCalledWith(true);
    });
    it('call toggleModal isError = true', () => {
      const component = start().buildComponent();
      isError = true;
      component.setProps({isError: isError});
      expect(toggleModal).toHaveBeenCalledWith(true);
    });
  });
  describe('setModal', () => {
    start();
    it('modal === ModalTypes.info', () => {
      const component = start().buildComponent();
      const instance = component.instance() as PasswordRestoreComponent;
      const modal = instance.setModal();
      expect(modal).toBe(ModalTypes.info);
    });
    it('modal === ModalTypes.error', () => {
      isError = true;
      const component = start().buildComponent();
      const instance = component.instance() as PasswordRestoreComponent;
      const modal = instance.setModal();
      expect(modal).toBe(ModalTypes.error);
    });
    it('modal === ModalTypes.success', () => {
      isLoad = true;
      const component = start().buildComponent();
      const instance = component.instance() as PasswordRestoreComponent;
      const modal = instance.setModal();
      expect(modal).toBe(ModalTypes.success);
    });
    it('modal !== ModalTypes.success', () => {
      isLoad = true;
      resultCode = 32;
      const component = start().buildComponent();
      const instance = component.instance() as PasswordRestoreComponent;
      const modal = instance.setModal();
      expect(modal).toBe(ModalTypes.info);
    });
  })
});


