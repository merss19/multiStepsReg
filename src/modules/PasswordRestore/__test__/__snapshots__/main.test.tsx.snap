// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PasswordRestore renders correctly 1`] = `
<div
  className="entry--min"
>
  <ReduxForm
    onSubmit={[Function]}
  />
  <Connect(CustomModalComponent)
    clickHandler={[Function]}
    isLoader={false}
    modal="info"
    resultText="resultText"
  />
</div>
`;
